plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -405,12 +405,34 @@ private function process_select_field( $field_data, $options ) {
 	private function process_checkbox_field( $field_data, $options ) {
 		$id = $options['id'];
 
-		// Checkbox fields are always optional. Log a warning if it's set explicitly as true.
-		$field_data['required'] = false;
+		if ( isset( $options['required'] ) && null === filter_var( $options['required'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE ) ) {
+			$message = sprintf( 'The required property for field with id: "%s" must be a boolean, you passed %s. The field will not be registered.', $id, gettype( $options['required'] ) );
+			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '9.8.0' );
+			return false;
+		}
 
-		if ( isset( $options['required'] ) && true === $options['required'] ) {
-			$message = sprintf( 'Registering checkbox fields as required is not supported. "%s" will be registered as optional.', $id );
-			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '8.6.0' );
+		if ( isset( $options['required'] ) ) {
+			// Filter the boolean in case the developer added it as a string.
+			$options['required']    = filter_var( $options['required'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );
+			$field_data['required'] = $options['required'];
+		}
+
+		if ( ( ! isset( $options['required'] ) || false === $options['required'] ) && ! empty( $options['error_message'] ) ) {
+			$message = sprintf( 'Passing an error message to a non-required checkbox "%s" will have no effect. The error message has been removed from the field.', $id );
+			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '9.8.0' );
+			unset( $field_data['error_message'] );
+		}
+
+		if ( isset( $options['error_message'] ) && ! is_string( $options['error_message'] ) ) {
+			$message = sprintf( 'The error_message property for field with id: "%s" must be a string, you passed %s. A default message will be shown.', $id, gettype( $options['error_message'] ) );
+			_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), '9.8.0' );
+			unset( $field_data['error_message'] );
+		}
```

Comment:
This is odd, isn't the $field_data already carrying defaults? We check for $options, but I don't see where the value from it gets copied to $field_data